jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -682,7 +728,9 @@ def model_image(scidata_bkg, scierr, scimask, refmask, ref_files, box_weights,
         model = np.where(already_modeled, 0., model)
 
         # Add to tracemodels
+        both_nan = np.isnan(tracemodels[order_str]) & np.isnan(model)
         tracemodels[order_str] = np.nansum([tracemodels[order_str], model], axis=0)
+        tracemodels[order_str][both_nan] = np.nan
```

Comment:
Previously this was leading to a situation where null values in trace were set to NaN for order 1 and zero for order 2. The new code ensures that null values in order 2 are set to NaN also.  (nansum sets all-NaN slices to zero)